go/libraries/doltcore/env/actions/infer_schema.go

```diff
@@ -181,7 +182,25 @@ func leastPermissiveType(strVal string, floatThreshold float64) typeinfo.TypeInf
 		return typeinfo.BoolType
 	}
 
-	return typeinfo.StringDefaultType
+	if strings.Contains(strVal, "{") {
+		var jsonMap map[string]interface{}
+		err := json.Unmarshal([]byte(strVal), &jsonMap)
+		if err == nil {
+			return typeinfo.JSONType
+		}
+	} else if strings.Contains(strVal, "[") {
```

Comment:
I *think* you may want to remove the `else` here and just have this be a separate, free-standing `if` block. 

Otherwise... if someone uses a JSON string like "[{a:'foo'}, {b:'bar'}]", it would attempt to parse it as a map and fail, and then not take the `else` path in the second block. 

Worth at least a quick test to see how the code acts in that case. 

Alternatively... you could probably collapse these two if blocks into one that contains `{` or `[` and just pass in a `interface{}` for the second arg to Unmarshall, especially since you don't need to look at the unmarshalled value.